zeeguu_api/api/topics.py

```diff
@@ -0,0 +1,179 @@
+from .utils.json_result import json_result
+from . import api
+
+session = zeeguu.db.session
+
+SUBSCRIBE_TOPIC = "subscribe_topic"
+UNSUBSCRIBE_TOPIC = "unsubscribe_topic"
+SUBSCRIBED_TOPICS = "subscribed_topics"
+UNFOLLOWED_TOPICS = "unfollowed_topics"
+FILTER_TOPIC = "filter_topic"
+UNFILTER_TOPIC = "unfilter_topic"
+FILTERED_TOPICS = "filtered_topics"
+
+
+# ---------------------------------------------------------------------------
+@api.route(f"/{SUBSCRIBE_TOPIC}", methods=("POST",))
+# ---------------------------------------------------------------------------
+@cross_domain
+@with_session
+def subscribe_to_topic_with_id():
+    """
+    :param: topic_id -- the id of the topic to be subscribed to.
+    Subscribe to the topic with the given id
+
+    :return: "OK" in case of success
+    """
+
+    topic_id = int(request.form.get('topic_id', ''))
+
+    topic_object = Topic.find_by_id(topic_id)
+    TopicSubscription.find_or_create(session, flask.g.user, topic_object)
+
+    return "OK"
+
+
+# ---------------------------------------------------------------------------
+@api.route(f"/{UNSUBSCRIBE_TOPIC}/<topic_id>", methods=("GET",))
+# ---------------------------------------------------------------------------
+@cross_domain
+@with_session
+def unsubscribe_from_topic(topic_id):
+    """
+    A user can unsubscribe from the topic with a given ID
+    :return: OK / ERROR
+    """
+
+    try:
+        to_delete = TopicSubscription.with_topic_id(topic_id, flask.g.user)
+        session.delete(to_delete)
+        session.commit()
+    except Exception as e:
+        return "OOPS. FEED AIN'T THERE IT SEEMS (" + str(e) + ")"
+
+    return "OK"
+
+
+# ---------------------------------------------------------------------------
+@api.route(f"/{SUBSCRIBED_TOPICS}", methods=("GET",))
+# ---------------------------------------------------------------------------
+@cross_domain
+@with_session
+def get_subscribed_topics():
+    """
+    A user might be subscribed to multiple topics at once.
+    This endpoint returns them as a list.
+
+    :return: a json list with feeds for which the user is registered;
+     every feed in this list is a dictionary with the following info:
+                id = unique id of the topic;
+                title = <unicode string>
+    """
+    subscriptions = TopicSubscription.all_for_user(flask.g.user)
+    topic_list = []
+    for sub in subscriptions:
+        try:
+            topic_list.append(sub.topic.as_dictionary())
+        except Exception as e:
+            zeeguu.log(str(e))
+
+    return json_result(topic_list)
+
+
+# ---------------------------------------------------------------------------
+@api.route(f"/{UNFOLLOWED_TOPICS}", methods=("GET",))
+# ---------------------------------------------------------------------------
+@cross_domain
+@with_session
+def get_unfollowed_topics():
+    """
+    Get a list of interesting topics for the given language.
```

Comment:
often the first words of the comment are
a better name for the function than the actual
function name .e.g. `get_interesting_topics` ...
what say you?

Reply:
Changed